doltpy/sql/sync/mysql.py

```diff
@@ -27,6 +27,6 @@ def get_target_writer(engine: Engine, update_on_duplicate: bool = True) -> DoltA
 
 def upsert_helper(table: Table, data: List[dict]):
     insert_statement = insert(table).values(data)
-    update_dict = {el.name: el for el in insert_statement.inserted if not el.primary_key}
+    update_dict = {el.name: el for el in insert_statement.inserted}
```

Comment:
Maybe @oscarbatori would be able to comment on the original code -- is there any unexpected behavior when we drop this primary key filter? Should we instead have a short circuit return if update_dict is empty? I have not spent a lot of time with the SQL-sync code since joining, can look at this more tomorrow.

Reply:
Thanks for your advice. I tried a short circuit return, but the code I wrote was not clean enough, and that lead to this one line change.